### PR TITLE
Fix callouts on scroll and resize

### PR DESCRIPTION
**Pull request by molly-moen**
This PR fixes a couple issues with callouts:
- In Google Blockly labs, we weren't updating callout position at all on scroll or resize. This is fixed by adding an event handler on both the main workspace and toolbox workspace viewport changed event, and adding an event when the browser itself resizes.
- For both Google Blockly and CDO Blockly, when a block was scrolled outside of the main workspace the would still show up floating on the page. We had logic to avoid this, but it was broken at some point. Now we check that the block is still inside the workspace, and if it's not we hide it.

### Before (CDO Blockly)
https://github.com/code-dot-org/code-dot-org/assets/33666587/7f310341-cb4c-4344-8583-f5d68908409c

### Before (Google Blockly)
https://github.com/code-dot-org/code-dot-org/assets/33666587/cf4ec0fb-a4be-4cb1-a05c-f69dc82eaff8

### After (CDO Blockly)

https://github.com/code-dot-org/code-dot-org/assets/33666587/20180ade-0333-4885-b14e-16f4f8d61330

### After (Google Blockly)

https://github.com/code-dot-org/code-dot-org/assets/33666587/a766f13b-1916-434a-af6b-b9b2c7ecdd33

## Links

- jira ticket: [CT-56](https://codedotorg.atlassian.net/browse/CT-56)

## Testing story
Tested locally.

## Follow-up work
For Google Blockly labs with the modal function editor enabled there is still one UI quirk remaining. The modal function editor has a duplicate toolbox, which means callouts on the toolbox are created for both the function editor and regular toolboxes. With the new and improved hiding logic these callouts do get hidden, but they briefly flicker in the top left corner on page load. I made a [follow up taks](https://codedotorg.atlassian.net/browse/CT-149) to see if we can solve this.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/callouts.js [hunk 1]
```diff
@@ -295,3 +301,15 @@ function reverseDirection(token) {
       return token;
   }
 }
+
+// Return true if the element is fully contained within the container.
+function elementIsInContainer(element, container) {
```
Comment (mikeharv) [inline apps/src/code-studio/callouts.js:306]: Nice!

File: apps/src/blockly/eventHandlers.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
+// Event Handlers for Google Blockly.
```
Comment (mikeharv) [inline apps/src/blockly/eventHandlers.js:1]: Is it bad that I didn't know this file existed? There are some other event handlers that we could potentially move here if we like this structure.

**Review by mikeharv — APPROVED** [review]
Looks good to me! I pulled and tested your changes, including running your level update script and confirmed that the callouts now track the block - and disappear when the block moves out of view. Two great improvements! 
Depending on how fast you scroll, I can get a slight "flicker" where the callout appears over the instructions for a single frame, but I'm not sure how hard that would be to fix. This solution seems very reasonable and should objectively improve the student experience.

File: apps/src/blockly/eventHandlers.js (hunk 2 above)
Comment (molly-moen) [reply apps/src/blockly/eventHandlers.js:1]: I added it recently :) I'm open to moving other event handlers in here! If we think we'll have a lot of event handlers we could also make it a folder to keep files a manageable length.

**Comment by molly-moen** [comment]
>  I can get a slight "flicker" where the callout appears over the instructions for a single frame.

@mikeharv my guess is that is just the events catching up with scrolling, it seems ok to me that it flickers briefly. I think that would be hard-ish to fix.

**Pull request merged**